cxx_argp_parser.h

```diff
@@ -53,19 +53,43 @@ class parser
 		};
 	}
 
-	/* for integers */
+	/* for signed integers */
 	template <typename T>
 	typename std::enable_if<
-	    std::numeric_limits<T>::is_integer,
+	    std::numeric_limits<T>::is_integer && std::numeric_limits<T>::is_signed,
 	    std::function<bool(const char *)>>::type
 	make_check_function(T &x)
 	{
 		const auto px = &x;
 		return [px](const char *arg) {
 			char *end;
-			*px = strtol(arg, &end, 0);
+			long long int value = strtoll(arg, &end, 0);
```

Comment:
Can't we use `auto` here and somehow use `strtoll` if `is_signed` and `strtoull` if not? With some template magic this should be possible, shouldn't it?

Reply:
yup, certainly possible!  as is also switching between between `strtol` and `strtoll` as appropriate.

I didn't think it was worth it: more template magic tends to slow down compile times as well as quickly making code less readable.  I'd be tempted to leave it until there are more cases to handle.